languages/kotlin/docs/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,122 @@
+
+## Step 2: add .docs/instructions.md
+
+This file contains instructions for the exercise. It should explicitly explain what the user needs to do (define a method with the signature `X(...)` that takes an A and returns a Z), and provide at least one example usage of that function. If there are multiple tasks within the exercise, it should provide an example of each.
+
+## Step 3: add .docs/hints.md
+
+If the user gets stuck, we will allow them to click a button requesting a hint, which shows this file. We will softly discourage them using it. The file should contain both general and task-specific "hints". These hints should be enough to unblock almost any
+
+## Step 4: add .docs/after.md (optional):
+
+Once the user completes the exercise they will be shown this file, which gives them any bonus information or further reading about the concept taught.
+
+These files are also all described in the [concept exercises document][docs-concept-exercises].
+
+## Step 5: add .meta/config.json
+
+This file contains metadata for the exercise. The metadata includes paths to the test file and the solution files, as well as a list of all the tests and the method they are calling.
+
+## Step 6: update languages/kotlin/config.json
+
+An entry should be added to the track's `config.json` file for the new concept exercise:
+
+```json
+{
+  ...
+  "exercises": {
+    "concept": [
+      ...
+      {
+        "slug": "<exercise-slug>",
+        "uuid": "<UUID>",
+        "concepts": ["<CONCEPT-1>", "<CONCEPT-2>"],
+        "prerequisites": ["<PREREQUISITE-1>", "<PREREQUISITE-2>"]
+      }
+    ]
+  }
+}
+```
+
+Where:
+
+- `<UUID>` - is unique exercise uuid;
+- `<CONCEPT-1`/`<CONCEPT-2>` - concepts that are explained in the exercise (at least one concept is required, more concepts are optional);
+- `<PREREQUISITE-1>`/`<PREREQUISITE-2>` - this exercises will be blocked for user unless all concepts from from `prerequisites` are already familiar for user (e.g. he finished exercises covering this concepts).
+
+## Step 7: adding track-specific files
+
+Having added the files that are not specific to the track, now is the time to create the track-specific files. These file will include:
+
+- A stub implementation file (`src/main/kotlin/<ExerciseSlug>.kt`).
+- A file containing the test suite (`src/test/kotlin/<ExerciseSlug>.kt`).
+- An example implementation file that passes all the tests (`meta/Example.kt`).
+
+What these files look like depends on your track. Note that some tracks might require more files in addition to the three files just mentioned.
+
+## Step 8: add analyzer (optional)
+
+Some exercises could benefit from having an exercise-specific analyzer. If so, please check the track's analyzer document for details on how to do this.
```

Comment:
Can we provide here a link to the analyzer docs mentioned?

Reply:
I'd rather mark this as a draft as we don't have proper analyzer for now.